R/vector_logic_linter.R

```diff
@@ -11,6 +11,14 @@
 #'   `&&` / `||` are not perfect substitutes because `&` is doing
 #'   method dispatch in an incompatible way.
 #'
+#' Moreover, be wary of code that may have side effects, most commonly
+#'   assignments. Consider `if ((a <- foo(x)) | (b <- bar(y))) { ... }`
+#'   vs. `if ((a <- foo(x)) || (b <- bar(y))) { ... }`. Because `||` exits
+#'   early, if `a` is `TRUE`,  the second condition will never be evaluated
+#'   and `b` will not be assigned. Such usage is not allowed by the Tidyverse
+#'   style guide, and the code can easily be refactored by pulling the
+#'   assignment outside of the condition, so using `||` is still preferable.
```

Comment:
```suggestion
#'   assignment outside the condition, so using `||` is still preferable.
```

Reply:
indeed looks like the more common choice 

https://books.google.com/ngrams/graph?content=outside+the%2Coutside+of+the&year_start=1800&year_end=2019&corpus=26&smoothing=3&case_insensitive=true

Reply:
Probably pull sth out of sth would be more correct.